### PR TITLE
fix(journal): Correctly handle client socket closing during compaction

### LINKED ISSUE #4839: Journal compaction: Outbound sockets not handled
Outbound sockets are currently not properly recorded in the  compactor, and thus are not properly pruned from compacted journals, which can cause issues.

### PR DESCRIPTION
**Pull request by theduke**
Previously outbound sockets were not properly recorded as open in the
compactor, and thus closed sockets were not properly removed from the
journal.

This change implements tracking for sockets after connecting to them.

Closes #4839

### CODE REVIEW

File: lib/journal/src/concrete/compacting.rs [hunk 1]
```diff
@@ -467,7 +467,7 @@ impl WritableJournal for CompactingJournalTx {
                 //  thus the entire branch of events it represents is discarded)
                 let mut erase = false;
                 let lookup = if matches!(&entry, JournalEntry::CloseFileDescriptorV1 { .. }) {
-                    if state.open_sockets.remove(fd).is_some() {
+                    if dbg!(state.open_sockets.remove(fd).is_some()) {
```
Comment (syrusakbary) [inline lib/journal/src/concrete/compacting.rs:470]: Undo `dbg!` ?

**Review by john-sharratt — COMMENTED** [review]
Did you mean to comment out all unit tests?

File: lib/journal/src/concrete/compacting.rs (hunk 1 above)
Comment (theduke) [reply lib/journal/src/concrete/compacting.rs]: Done

**Comment by theduke** [comment]
> Did you mean to comment out all unit tests?

They were already commented out, I un-commented the runner and added a new test.

Followup should fix up the other tests.
I had a brief look, but they require some work.

**Review by syrusakbary — APPROVED** [review]

**Comment by theduke** [comment]
@john-sharratt for posterity, I phrased it a bit incorrectly: before, non of the tests were running because they were gated behind the `journal` feature flag, which is probably a copy paste leftover (from the wasix crate).

The journal crate doesn't have a `journal` feature flag, so the whole test module was inactive.

**Pull request merged**